Python/Python-tokenize.c

```diff
@@ -177,14 +178,71 @@ _tokenizer_error(struct tok_state *tok)
     return result;
 }
 
+static PyObject *
+_get_current_line(tokenizeriterobject *it, const char *line_start, Py_ssize_t size)
+{
+    _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(it);
+    PyObject *line;
+    if (it->tok->lineno != it->last_lineno) {
+        // Line has changed since last token, so we fetch the new line and cache it
+        // in the iter object.
+        Py_XDECREF(it->last_line);
+        line = PyUnicode_DecodeUTF8(line_start, size, "replace");
+        it->last_line = line;
+        if (it->tok->lineno != it->last_end_lineno) {
+            it->byte_col_offset_diff = 0;
+        }
+    }
+    else {
+        // Line hasn't changed so we reuse the cached one.
+        line = it->last_line;
+    }
+    return line;
+}
```

Comment:
Nit: feel free to ignore

```suggestion
    PyObject *line = it->last_line;
    if (it->tok->lineno != it->last_lineno) {
        // Line has changed since last token, so we fetch the new line and cache it
        // in the iter object.
        Py_XDECREF(it->last_line);
        line = PyUnicode_DecodeUTF8(line_start, size, "replace");
        it->last_line = line;
        if (it->tok->lineno != it->last_end_lineno) {
            it->byte_col_offset_diff = 0;
        }
    }
    return line;
}
```